### PR TITLE
Add new man pages for rnp/rnpkeys/librnp.

### PR DESCRIPTION
**Pull request by ni4**
This PR adds new man pages for the RNP suite instead of outdated NetPGP ones.
Text of the man pages is now stored in markdown, and rendered to roff via pandoc.
Currently this is done manually since it seems to be an overload to add pandoc calls to the CMake files.
Also man pages are now copied during the install step.

Feel free to add any suggestions/corrections/share experience about the man pages authoring.

### CODE REVIEW

**Comment by ni4** [comment]
@dkg would be thankful if you have time to take a look on this.

**Comment by dkg** [comment]
@ni4, this looks great!  I haven't tried rendering the manpages directly with troff or anything, but the overall approach seems reasonable to me.

I'm not enough of a cmake guru to tell you how to do it, but it might be nice to include some sort of optional build step that would rebuild the roff files from the markdown.  I understand that you don't want to introduce a build dependency on pandoc, but for debian packaging, i'll probably re-generate the roff files from markdown during build anyway.  If you've got a particular way that you're preferring to invoke it, recording that invocation someplace in the build tooling will make it easier for downstream folks like me (and maybe for future upstream maintainers too).

thanks for doing this!

**Comment by ronaldtse** [comment]
@ni4 I'd prefer not to include the rendered man pages in git but always re-generate them from source.

Re: packaging, I believe packagers should use published releases instead of git head, so it's not that much of an issue for them (the published release should have the rendered man pages).

**Comment by dkg** [comment]
Just to clarify: I'm only one datapoint, but as a packager, i'm going to do my best to re-generate things using packaged tooling based on the "preferred form of modification" -- meaning, the underlying source.  so if you ship generated nroff and markdown, i'll probably ignore/discard the nroff and rebuild from the markdown.  I think (i hope) that's a good thing for both RNP and Debian and its downstream users.

I agree with @ronaldtse that it's generally not good form to commit generated files to an active revision control system -- better to only commit the source and the build instructions.

As for packagers using "published releases", where would we get them from that *aren't* based on the contents of the git repo?  The only place i've seen that i can get official "releases" from is github, which are just tagged releases.  Even the Download link on https://www.rnpgp.com/software/rnp/ redirects to the github tag page.

If there's some other place to retrieve published releases, i'd appreciate a pointer.

**Comment by ronaldtse** [comment]
Thanks @dkg. For published releases, I just meant our GitHub releases page. It is possible to provide a slightly processed release archive that includes compiled man pages, but perhaps that's not even necessary...

**Comment by ni4** [comment]
Thanks for the input. I'd also stick to 'have just source' strategy, it's quite logical. The only concern is the need to install `pandoc` for every rnp user to have the man pages, isn't this an overkill?

**Comment by dkg** [comment]
Is it possible that you can use pandoc to generate html variants of the manpages, and then publish them automatically upon each release in a known URL?  that way, casual users-from-source can build and install rnp without worrying about pandoc (they can refer to the local .md files that they can read directly, or they can read them on the web), and packagers that are able to depend on a functional OS (most packagers) can just build-depend on pandoc.

note that this might mean needing some sort of build flag that controls whether the manpages are built or not (or autodetection based on the presence of pandoc), so that casual users-from-source don't end up with build errors due to a missing pandoc installation.

**Comment by dkg** [comment]
(just wanted to note that users of packaged versions of rnp will *not* need pandoc installed to read the manpages, in case that wasn't clear, just like they won't need a C compiler to use `rnp` -- pandoc is only a build-dependency, not a runtime dependency)

**Comment by ni4** [comment]
@dkg Thanks, addded cmake module/function to build man pages depending on pandoc availability, and issuing a warning if it is not available.

**Review by ronaldtse — APPROVED** [review]
Thanks @ni4!

**Comment by ronaldtse** [comment]
Sorry for the lack of discussion but I've converted the man page flow to use AsciiDoc instead of Markdown.

Somehow the builds are still showing:
```
CMake Warning at cmake/Modules/AdocMan.cmake:47 (message):
  asciidoctor not found, man pages will not be generated.  Install
  asciidoctor or use the CMAKE_PROGRAM_PATH variable.
```

But we could first merge this and fix the CI in the #1437.

File: cmake/Modules/AdocMan.cmake
Comment (ni4) [inline cmake/Modules/AdocMan.cmake:99]: @ronaldtse Should not we have `- 5` here? Currently it generates file like `rnp.1.a`.

Comment (ronaldtse) [reply cmake/Modules/AdocMan.cmake:99]: Ha, I totally missed this function. Will fix (or if you can fix it...)

Comment (ni4) [reply cmake/Modules/AdocMan.cmake:99]: @ronaldtse No problem, I'll take care of it. Btw, what about to create separate repository (if there is no one yet) and put PandocMan.cmake, AdocMan.cmake there? I think people might find that helpful for other projects.

Comment (ronaldtse) [reply cmake/Modules/AdocMan.cmake]: I've just fixed it in ff551cd8

**Comment by ronaldtse** [comment]
All builds are passing. @antonsviridenko @rrrooommmaaa can we have one more approval and merge? Thanks.

File: cmake/Modules/AdocMan.cmake
Comment (ronaldtse) [reply cmake/Modules/AdocMan.cmake]: > what about to create separate repository (if there is no one yet) and put PandocMan.cmake, AdocMan.cmake there? 

@ni4 this is a good idea. Let's do that.

**Review by antonsviridenko — APPROVED** [review]

File: cmake/Modules/AdocMan.cmake
Comment (ni4) [reply cmake/Modules/AdocMan.cmake]: > I've just fixed it in ff551cd

Thanks!

> @ni4 this is a good idea. Let's do that.

@ronaldtse  Do I have access to create a new repository (and what would be good name for it? `cmake-modules`?), or you'll do that? Also we may move `version.cmake` in that repository for less clutter.

Comment (ronaldtse) [reply cmake/Modules/AdocMan.cmake]: Of course you do, but I've just made it 😉  https://github.com/rnpgp/cmake-modules It's all yours!

Comment (ni4) [reply cmake/Modules/AdocMan.cmake]: @ronaldtse Thanks! :-)

File: src/lib/librnp.3.adoc
Comment (ni4) [inline src/lib/librnp.3.adoc:37]: @ronaldtse Does this came here by mistake or by intention :-) ?

Comment (ronaldtse) [reply src/lib/librnp.3.adoc:37]: Complete mistake. Sorry. Can you help fix it?

Comment (ronaldtse) [reply src/lib/librnp.3.adoc:37]: Thank you for spotting...

Comment (ni4) [reply src/lib/librnp.3.adoc:37]: @ronaldtse Sure.

**Comment by eli-schwartz** [comment]
> Just to clarify: I'm only one datapoint, but as a packager, i'm going to do my best to re-generate things using packaged tooling based on the "preferred form of modification" -- meaning, the underlying source. so if you ship generated nroff and markdown, i'll probably ignore/discard the nroff and rebuild from the markdown. I think (i hope) that's a good thing for both RNP and Debian and its downstream users.



> Is it possible that you can use pandoc to generate html variants of the manpages, and then publish them automatically upon each release in a known URL? that way, casual users-from-source can build and install rnp without worrying about pandoc (they can refer to the local .md files that they can read directly, or they can read them on the web), and packagers that are able to depend on a functional OS (most packagers) can just build-depend on pandoc.
> 
> note that this might mean needing some sort of build flag that controls whether the manpages are built or not (or autodetection based on the presence of pandoc), so that casual users-from-source don't end up with build errors due to a missing pandoc installation.


@dkg, just to note this is hostile to Gentoo, a from-source distro. Compiling pandoc is extremely painful and not even possible at all on many arches where we don't have GHC available. We don't want people to spend days just compiling manpages.

asciidoctor is not much better. ruby is unpopular, ruby *gems* have deep dependency trees, it adds considerable burden.

Gentoo also vehemently disputes "preferred form of modification", so you really should not go around telling upstream projects that a Debian policy is globally representative of all distros.

To put it bluntly, I only just realized today, you are personally the reason my system is missing the manpages for this package and I find it a bit... frustrating.

**Comment by eli-schwartz** [comment]
> Sorry for the lack of discussion but I've converted the man page flow to use AsciiDoc instead of Markdown.

@ronaldtse, BTW in general I don't mind asciidoc, but the canonical tool for it is "asciidoc": https://github.com/asciidoc-py/asciidoc-py/

It is significantly less burdensome to install than the ruby based asciidoctor. This would be less of a problem if proper release tarballs were provided of course.

**Pull request merged**
